destination/schema/snowflake/datatype_test.go

```diff
@@ -0,0 +1,145 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snowflake
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/matryer/is"
+)
+
+func TestDataTypeContainer_JSONUnmarshal(t *testing.T) {
+	testCases := []struct {
+		have string
+		want DataType
+	}{{
+		have: `{"type":"FIXED","precision":38,"scale":0,"nullable":true}`,
```

Comment:
The test cases are taken from the [Snowflake documentation page](https://docs.snowflake.com/en/sql-reference/sql/show-columns#examples), except the last couple of cases, which are not described in docs. For those I have created a test table in snowflake and checked the output of `SHOW COLUMNS`.